lib/components/channel_panel.dart

```diff
@@ -99,8 +121,22 @@ class ChannelPanelWidget extends StatelessWidget {
           return Padding(
             padding: const EdgeInsets.symmetric(horizontal: 16),
             child: Row(children: [
+              IconButton(
+                  onPressed: () {
+                    if (_isEmotePickerVisible) {
+                      hideEmotePicker();
```

Comment:
I would just inline these functions, and you can also do:

```
setState(() {
  _isEmotePickerVisible = !_isEmotePickerVisible;
});
```

Reply:
I guess a toggle function would work, and I'd have to conditionally call it on the inputs `onTap`

Reply:
Can we call

```
onTap: () => setState(() => _isEmotePickerVisible = false),
```

in the handler? I think it's slightly better to just inline these as it's more explicit about what's happening.